### PR TITLE
[Doc] Update the Data Fetching documentation

### PR DESCRIPTION
**Pull request by fzaninotto**
## Problem

I always struggle to find information about the data provider in the doc

## Solution

- less chapters
- more section titles
- reorganize sections for clarity

### CODE REVIEW

File: docs/DataProviders.md [hunk 1]
```diff
@@ -1,14 +1,60 @@
...
+dataProvider
+    .getOne('posts', { id: 123 })
+    .then(response => {
+        console.log(response.data); // { id: 123, title: "hello, world" }
+    });
+```
+
+It's the Data Provider's job to turn these method calls into HTTP requests, and transform the HTTP responses to the data format expected by react-admin. In technical terms, a Data Provider is an *adapter* for an API. 
+
+Thanks to this adapter system, react-admin can communicate with any API, whether it uses REST, GraphQL, RPC, or even SOAP, regardless of the dialect it uses. Check out the [list of supported backends](./DataProviderList.md) to pick an open-source package for your API.
```
Comment (adguernier) [inline docs/DataProviders.md:28]: This paragraph is almost a repetition of the first paragraph of the introduction

```suggestion
Thanks to this adapter system, react-admin can communicate with any API. Check out the [list of supported backends](./DataProviderList.md) to pick an open-source package for your API.
```

File: docs/DataProviderWriting.md [hunk 2]
```diff
@@ -9,106 +9,58 @@ APIs are so diverse that quite often, none of [the available Data Providers](./D
...
-    ids: [123, 234],
-    data: { views: 0 },
-});
-dataProvider.delete('posts', {
-    id: 123,
-    previousData: { title: "hello, world" }
-});
-dataProvider.deleteMany('posts', { ids: [123, 234] });
+**Interface**
+```tsx
```
[the remark this answers is not in the document]
Comment (fzaninotto) [reply docs/DataProviderWriting.md:41]: Right: it's just TypeScript. I don't know what to do in this case.

File: docs/EditTutorial.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ To better understand how to use the various react-admin hooks and components ded
 
 ### An Edition View Built By Hand
 
-Here is how you could write a book edition view in pure React, leveraging react-admin's [data fetching hooks](./DataProviderIntroduction.md), and [react-hook-form](https://react-hook-form.com/) to bind form inputs with a record object:
+Here is how you could write a book edition view in pure React, leveraging react-admin's [data fetching hooks](./Actions.md), and [react-hook-form](https://react-hook-form.com/) to bind form inputs with a record object:
```
[the remark this answers is not in the document]
Comment (slax57) [reply docs/EditTutorial.md:22]: Actually I realize that you did not really rename any files but rather move some sections around in already existing files, so no other choice I guess...

**Comment by fzaninotto** [comment]
Rebased, the JS/TS problem is fixed, and the review is applied.

**Review by slax57 — APPROVED** [review]

**Pull request merged**